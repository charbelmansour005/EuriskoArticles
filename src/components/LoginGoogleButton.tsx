import GoogleLogo from '../../assets/google.png';
import {StyleSheet, Text, View, Image, Pressable} from 'react-native';
import {themeColors} from '../helpers/themeColors';
import React from 'react';

const LoginGoogleButton = (): JSX.Element => {
  return (
    <>
      <Pressable
        android_disableSound={true}
        android_ripple={{color: themeColors.salmon, borderless: false}}
        style={styles.GoogleBtn}>
        <View style={styles.rowflex}>
          <Image source={GoogleLogo} style={{width: 30, height: 30}} />
          <Text style={styles.GoogleText}>Sign in with Google</Text>
        </View>
      </Pressable>
      <View style={{marginTop: 20, ...styles.rowflex}}>
        <Text style={{color: '#131938'}}> Don't have an account?</Text>
        <Text style={{color: '#131938', fontWeight: 'bold'}}>{` Sign up`}</Text>
      </View>
    </>
  );
};

export default LoginGoogleButton;

const styles = StyleSheet.create({
  GoogleBtn: {
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: themeColors.grey,
    padding: 8,
    width: 'auto',
    maxWidth: '100%',
    marginTop: 25,
    borderRadius: 5,
  },
  rowflex: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  GoogleText: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 15,
    textAlign: 'center',
    letterSpacing: 1,
    marginLeft: 14,
  },
});
