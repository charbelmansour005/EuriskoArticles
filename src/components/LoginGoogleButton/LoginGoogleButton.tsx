import GoogleLogo from '../../../assets/google.png'
import {StyleSheet, Text, View, Image, Pressable} from 'react-native'
import {themeColors} from '../../helpers/themeColors'
import React from 'react'
// import {useAppSelector} from '../../app/rtkHooks'
import {Fragment} from 'react'

interface Props {
  SignInToast: () => void
  SignUpToast: () => void
}

const LoginGoogleButton = ({...props}: Props) => {
  // const language = useAppSelector(state => state.language)
  return (
    <Fragment>
      <Pressable
        onPress={props.SignInToast}
        testID="googlePressable"
        android_disableSound={true}
        android_ripple={{color: themeColors.white, borderless: false}}
        style={styles.GoogleBtn}>
        <View style={styles.rowflex} testID="btnContainer">
          <Image
            source={GoogleLogo}
            style={{width: 30, height: 30}}
            testID="googleImage"
          />
          <Text style={styles.GoogleText} testID="googleText">
            Sign in with Google
          </Text>
        </View>
      </Pressable>
      <View style={{marginTop: 20, ...styles.rowflex}} testID="signUpContainer">
        <Text style={{color: 'white'}} testID="googleNoAcc">
          Don't have an account?
        </Text>
        <Text
          onPress={props.SignUpToast}
          style={{color: themeColors.skyblue, fontWeight: 'bold'}}
          testID="googleSignUp">
          {' '}
          Sign up
        </Text>
      </View>
    </Fragment>
  )
}

export default LoginGoogleButton

const styles = StyleSheet.create({
  GoogleBtn: {
    backgroundColor: '#5865F2',
    borderWidth: 1,
    borderColor: '#5865F2',
    padding: 5,
    width: '92%',
    maxWidth: '100%',
    marginTop: 25,
    borderRadius: 3,
    alignItems: 'center',
  },
  rowflex: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  GoogleText: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 15,
    textAlign: 'center',
    letterSpacing: 1,
    marginLeft: 14,
  },
})
