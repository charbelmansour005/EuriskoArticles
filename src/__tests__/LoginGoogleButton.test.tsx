import React from 'react'
import renderer from 'react-test-renderer'
import LoginGoogleButton from '../components/LoginGoogleButton/LoginGoogleButton'

describe('LoginGoogleButton Component', () => {
  test('renders correctly', async () => {
    const tree = renderer.create(<LoginGoogleButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
