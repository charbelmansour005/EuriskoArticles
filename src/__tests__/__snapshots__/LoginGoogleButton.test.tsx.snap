// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginGoogleButton Component renders correctly 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "lightgray",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginTop": 25,
        "maxWidth": "100%",
        "padding": 8,
        "width": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/google.png",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 15,
            "fontWeight": "bold",
            "letterSpacing": 1,
            "marginLeft": 14,
            "textAlign": "center",
          }
        }
      >
        Sign in with Google
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#131938",
        }
      }
    >
      Don't have an account?
    </Text>
    <Text
      style={
        Object {
          "color": "#131938",
          "fontWeight": "bold",
        }
      }
    >
      Sign up
    </Text>
  </View>,
]
`;

exports[`renders correctly 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "lightgray",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginTop": 25,
        "maxWidth": "100%",
        "padding": 8,
        "width": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/google.png",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 15,
            "fontWeight": "bold",
            "letterSpacing": 1,
            "marginLeft": 14,
            "textAlign": "center",
          }
        }
      >
        Sign in with Google
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#131938",
        }
      }
    >
      Don't have an account?
    </Text>
    <Text
      style={
        Object {
          "color": "#131938",
          "fontWeight": "bold",
        }
      }
    >
      Sign up
    </Text>
  </View>,
]
`;
